File: src/demo/c/testdll.c
#include <windows.h>

BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved) {
	switch (fdwReason) {
		case DLL_PROCESS_ATTACH:
			MessageBox(NULL, "Hello World", "Test!", MB_OK);
			break;

		case DLL_THREAD_ATTACH:
			break;

		case DLL_THREAD_DETACH:
			break;

		case DLL_PROCESS_DETACH:
			break;
	}

	return TRUE;
}

File: src/demo/rs/testdll.rs
use std::ptr::null_mut;
use winapi::{shared::{minwindef::{DWORD, HINSTANCE, LPVOID}, ntdef::{TRUE}}, um::{winnt::{BOOLEAN, DLL_PROCESS_ATTACH}, winuser::{MB_OK, MessageBoxA}}};

#[unsafe(no_mangle)]
extern "system" fn DllMain(_hinst_dll: HINSTANCE, fdw_reason: DWORD, _lpv_reserved: LPVOID) -> BOOLEAN {
    match fdw_reason {
		DLL_PROCESS_ATTACH => {
			unsafe { MessageBoxA(null_mut(), c"rusty Hello World".as_ptr(), c"Test!".as_ptr(), MB_OK) };
        }
        _ => {}
	}

	TRUE
}
